crates/sui-types/src/messages.rs

```diff
@@ -664,13 +674,13 @@ impl Transaction {
 
 impl Hash for Transaction {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.data.hash(state);
+        self.signed_data.data.hash(state);
```

Comment:
Should we not include the user signature here?

Reply:
+1

Reply:
This should be here.